core/src/queue.rs

```diff
@@ -147,49 +147,84 @@ impl Queue {
         tx: VersionedAcceptedTransaction,
         wsv: &WorldStateView,
     ) -> Result<(), (VersionedAcceptedTransaction, Error)> {
```

Comment:
Looks to me like the signature is a little off. Do you think it should be a tuple instead of `struct`?

Reply:
Sorry i'm not quite get your point on this one.

Reply:
Consider adding:
```rust 
struct Failure {
    tx: VersionedAcceptedTransaction, 
    err: Error
}
```

For clarity and ease of use.